### PR TITLE
wings: remove *ValkyrieBehavior model extensions 

### PR DESCRIPTION
**Pull request by no-reply**
these modules try to reimplement Hydra::PCDM and Hydra::Works directly on valkyrie models, injecting them at conversion time. bundling this logic, which is mostly concerned with querying, creates undesirable coupling between data representation and business concerns.

removing it should help clarify what the application needs to support to handle valkyrie objects natively.

```
Calculating -------------------------------------
convert a Monograph object to ActiveFedora
                         95.083  (± 8.4%) i/s -    477.000  in   5.053864s
convert a derived Valkyrie object back to ActiveFedora
                         47.188  (± 8.5%) i/s -    236.000  in   5.039478s
save a Valkyrie object
                          6.966  (±14.4%) i/s -     35.000  in   5.094816s
```

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by no-reply** [comment]
to keep scope from creeping and reviews from becoming more complex, i'm submitting fixes for the issues this PR reveals as separate patches: https://github.com/samvera/hyrax/pull/4639

File: lib/wings/orm_converter.rb [hunk 1]
```diff
@@ -48,15 +40,9 @@ def self.base_for(klass:)
     #   results in long methods
     def self.to_valkyrie_resource_class(klass:)
       relationship_keys = klass.respond_to?(:reflections) ? relationship_keys_for(reflections: klass.reflections) : []
-      relationship_keys.delete('member_ids')
```
Comment (no-reply) [inline lib/wings/orm_converter.rb:51]: handling these keys like any other, now.

File: lib/wings/orm_converter.rb [hunk 2]
```diff
@@ -48,15 +40,9 @@ def self.base_for(klass:)
...
-      relationship_keys.delete('member_of_collection_ids')
       reflection_id_keys = klass.respond_to?(:reflections) ? klass.reflections.keys.select { |k| k.to_s.end_with? '_id' } : []
 
       Class.new(base_for(klass: klass)) do
-        include Wings::CollectionBehavior if klass.included_modules.include?(Hyrax::CollectionBehavior)
```
Comment (no-reply) [inline lib/wings/orm_converter.rb:56]: this is the biggest change here. no longer injecting Hydra::Works-like behavior into valkyrie resources at runtime. now relying on the `Hyrax::FileSet`, `Hyrax::PcdmCollection` and `Hyrax::Work` classes to provide all the model layer behavior we want.

File: app/services/hyrax/thumbnail_path_service.rb [hunk 3]
```diff
@@ -22,6 +22,11 @@ def call(object)
 
       private
 
+      def audio?(thumb)
+        service = thumb.respond_to?(:audio?) ? thumb : Hyrax::FileSetTypeService.new(file_set: thumb)
```
Comment (no-reply) [inline app/services/hyrax/thumbnail_path_service.rb:26]: no longer adding `Hydra::Works::MimeTypes` to generated classes. a partial re-implementation is provided as `Hyrax::FileSetTypeService`

**Review by jeremyf — APPROVED** [review]

**Pull request merged**